src/jit/lsraxarch.cpp

```diff
@@ -2581,6 +2588,44 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
                 break;
             }
 
+            case NI_AVX2_GatherVector128:
+            case NI_AVX2_GatherVector256:
+            {
+                assert(numArgs == 3);
+                // Any pair of the index, mask, or destination registers should be different
+                srcCount += BuildOperandUses(op1);
+                srcCount += BuildDelayFreeUses(op2);
+
+                // get a tmp register for mask that will be cleared by gather instructions
+                buildInternalFloatRegisterDefForNode(intrinsicTree, allSIMDRegs());
+                setInternalRegsDelayFree = true;
+
+                buildUses = false;
+                break;
+            }
+
+            case NI_AVX2_GatherMaskVector128:
+            case NI_AVX2_GatherMaskVector256:
+            {
+                assert(numArgs == 5);
+                // Any pair of the index, mask, or destination registers should be different
```

Comment:
This comment mentions index, mask and destination; it needs to map those to the opN names. Also, what about the address? Doesn't that also need to be distinct from the target, as the target will be overwritten?

Reply:
Thanks, will update the comment.

> Also, what about the address? Doesn't that also need to be distinct from the target, as the target will be overwritten?

Don't you mean the address base register? If yes, that is a GPR and cannot be same as the vector target register.